src/lib/assets/yaml.ts

```diff
@@ -203,24 +185,29 @@ export function zarfYamlChart({ name, image, config }: Assets, path: string) {
       {
         name: "module",
         required: true,
-        charts: [
-          {
-            name: "module",
-            namespace: "pepr-system",
-            version: `${config.appVersion || "0.0.1"}`,
-            localPath: path,
-          },
-        ],
+        manifests: !chart ? [{ name: "module", namespace: "pepr-system", files: [path] }] : undefined,
+        charts: chart
+          ? [{ name: "module", namespace: "pepr-system", version: `${config.appVersion || "0.0.1"}`, localPath: path }]
+          : undefined,
         images: [image],
       },
     ],
   };
+}
 
-  return dumpYaml(zarfCfg, { noRefs: true });
+// Writes Zarf YAML for a manifest file
+export function zarfYaml(assets: Assets, path: string) {
+  return dumpYaml(generateZarfConfig(assets, path), { noRefs: true });
 }
 
+// Writes Zarf YAML for a Helm chart
+export function zarfYamlChart(assets: Assets, path: string) {
+  return dumpYaml(generateZarfConfig(assets, path, true), { noRefs: true });
+}
+
+// Generates and writes all resources into a YAML string
 export async function allYaml(assets: Assets, rbacMode: string, imagePullSecret?: string) {
-  const { name, tls, apiToken, path } = assets;
+  const { name, tls, apiToken, path, capabilities = [] } = assets; // Ensure default empty array for capabilities
```

Comment:
This seems due to a non-null assertion [here](https://github.com/defenseunicorns/pepr/blob/ec6922a27a3d464a83c6347c43e6554d4a746d84/src/lib/assets/index.ts#L28). Can we initialize `capabilities` to `[]` in that file so that we confine any initialization quirks to one spot?

Reply:
✅ Done